src/main/java/graphql/execution/incremental/DeferredExecutionSupport.java

```diff
@@ -155,25 +153,24 @@ private Supplier<CompletableFuture<DeferredFragmentCall.FieldWithExecutionResult
             return dfCache.computeIfAbsent(
                     currentField.getResultKey(),
                     // The same field can be associated with multiple defer executions, so
-                    // we memoize the field resolution to avoid multiple calls to the same data fetcher
-                    key -> FpKit.interThreadMemoize(() -> {
-                                CompletableFuture<FieldValueInfo> fieldValueResult = resolveFieldWithInfoFn
-                                        .apply(executionContext, callParameters);
-
-                                // Create a reference to the CompletableFuture that resolves an ExecutionResult
-                                // so we can pass it to the Instrumentation "onDispatched" callback.
-                                CompletableFuture<ExecutionResult> executionResultCF = fieldValueResult
-                                        .thenCompose(fvi -> fvi
-                                                .getFieldValueFuture()
-                                                .thenApply(fv -> ExecutionResultImpl.newExecutionResult().data(fv).build())
-                                        );
-
-                                return executionResultCF
-                                        .thenApply(executionResult ->
-                                                new DeferredFragmentCall.FieldWithExecutionResult(currentField.getResultKey(), executionResult)
-                                        );
-                            }
-                    )
+                    // we cache the field resolution to avoid multiple calls to the same data fetcher
+                    key -> {
```

Comment:
This is the main change - instead of creating suppliers that were being called just once the initial response was complete, we are executing the fields as they come and adding the CompletableFutures to the queue in `IncrementalCallState`